cl/search/tasks.py

```diff
@@ -385,20 +405,24 @@ def update_child_documents_by_query(
 
     :param self: The celery task
     :param es_document: The Elasticsearch Document type to update.
-    :param parent_instance: The parent instance containing the fields to update.
+    :param parent_instance_id: The parent instance ID containing the fields to update.
+    :param throttling_id: The throttling ID.
     :param fields_to_update: List of field names to be updated.
     :param fields_map: A mapping from model fields to Elasticsearch document fields.
     :return: None
     """
 
     s = es_document.search()
     main_doc = None
+    parent_instance = None
     if es_document is PositionDocument:
-        s = s.query("parent_id", type="position", id=parent_instance.pk)
-        main_doc = get_doc_from_es(PersonDocument, parent_instance.pk)
+        s = s.query("parent_id", type="position", id=parent_instance_id)
+        main_doc = get_doc_from_es(PersonDocument, parent_instance_id)
```

Comment:
We should use the `exists` method instead of `get` here. The `exists` method uses  a HEAD request which returns only the headers and not the entire message-body in the response. This approach is faster because it transfers less data in the request.

We only check the value of `main_doc` in line 427 without using the response data.

Reply:
Yes, you're correct. We can use `exists` instead of `get` here since we don't actually need the document instance. I initially used the document instance to access the prepare method, but we can also access the prepare method simply by instantiating the document class as `PersonDocument()` without needing the document instance itself.

I've also implemented this improvement in the signal processor by renaming `get_or_create_doc` to `exists_or_create_doc`, so it now uses exists instead of get. Additionally, I've updated the methods where the document instances were previously used for the prepare method to now only instantiate the document class.

Thanks!